src/iohumdrum.cpp

```diff
@@ -11564,7 +11563,11 @@ void HumdrumInput::addTextElement(
     std::vector<std::string> pieces;
     hre.split(pieces, data, "\\\\n");
 
-    for (int i = 0; i < (int)pieces.size(); i++) {
+    const size_t piecesSize = pieces.size();
```

Comment:
By the logic I give for using `int` in the comment for line 11597, your fix is proper for long-term stability of the code.

However, note that I claim that `pieces` can never have a size less than one, so there is a guarantee that no memory leak can occur (which would happen if `pieces` had a size of 0).  In other words, once this function is entered, it is committed to inserting at least one text element, even if it were an empty string.

As an aside: the for-loop and the lines:

```
    hum::HumRegex hre;
    std::vector<std::string> pieces;
    hre.split(pieces, data, "\\\\n");
```

Are a fairly recent addition to the function to allow for line breaks in the text (mostly for `<dir>` elements).  A string such as `line1\nline2` will be converted into two text elements containing `line1` and `line2` with an `<lb>`  between them.  So `pieces` will have a size of 1 if there are not line breaks, and otherwise will be greater than one if there are any line breaks indicated by `\n` characters in the input string.

Reply:
How `hum::HumRegex::split` behaves in case of empty string? I suppose in this case returned vector will be empty. Or may contain empty string as the only element. Both implementations seem logical.

May be you are right and leak will never happen. To say that for sure one should know if `hum::HumRegex::split` ever returns empty vector.

Proposed change makes code work properly despite of logic of `hum::HumRegex::split`.

Think of possible case when logic of `hum::HumRegex::split` changes. Then here real leak might happen.

Reply:
@craigsapp what do you want to do with this?